synapse/federation/federation_server.py

```diff
@@ -148,6 +148,30 @@ def _handle_incoming_transaction(self, origin, transaction, request_time):
 
         logger.debug("[%s] Transaction is new", transaction.transaction_id)
 
+        # Reject if PDU count > 50 and EDU count > 100
+        if (len(transaction.pdus) > 50
+                or (hasattr(transaction, "edus") and len(transaction.edus) > 100)):
+            response = {
+                "pdus": {}
+            }
+
+            for pdu_key in transaction["pdus"].keys():
+                response["pdus"][pdu_key] = {
+                    "error": "Processing failed. More than 50 PDUs or 100 EDUs sent."
+                }
+
+            logger.debug(
```

Comment:
this probably wants to be at least an info (and not contain a mahoosive list of events)